src/main/java/com/wire/bots/roman/resources/BroadcastV2Resource.java

```diff
@@ -0,0 +1,209 @@
+package com.wire.bots.roman.resources;
+
+import com.codahale.metrics.annotation.Metered;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wire.bots.roman.DAO.BotsDAO;
+import com.wire.bots.roman.DAO.BroadcastDAO;
+import com.wire.bots.roman.Sender;
+import com.wire.bots.roman.filters.ServiceTokenAuthorization;
+import com.wire.bots.roman.model.BroadcastMessage;
+import com.wire.bots.roman.model.Report;
+import com.wire.lithium.server.monitoring.MDCUtils;
+import com.wire.xenon.assets.*;
+import com.wire.xenon.backend.models.ErrorMessage;
+import com.wire.xenon.tools.Logger;
+import io.swagger.annotations.*;
+import org.jdbi.v3.core.Jdbi;
+
+import javax.annotation.Nullable;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.*;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Base64;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.logging.Level;
+
+import static com.wire.bots.roman.Const.APP_KEY;
+import static com.wire.bots.roman.Const.PROVIDER_ID;
+
+@Api
+@Path("/broadcast/v2")
+@Produces(MediaType.APPLICATION_JSON)
+public class BroadcastV2Resource {
+    private final Sender sender;
+    private final BotsDAO botsDAO;
+    private final BroadcastDAO broadcastDAO;
+    private final ExecutorService broadcast;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public BroadcastV2Resource(Jdbi jdbi, Sender sender, ExecutorService broadcast) {
+        this.sender = sender;
+        this.broadcast = broadcast;
+
+        botsDAO = jdbi.onDemand(BotsDAO.class);
+        broadcastDAO = jdbi.onDemand(BroadcastDAO.class);
+    }
+
+    @POST
+    @ApiOperation(value = "Broadcast message on Wire", authorizations = {@Authorization(value = "Bearer")})
+    @ApiResponses(value = {@ApiResponse(code = 401, message = "Not authenticated", response = Report.class)})
+    @Metered
+    @ServiceTokenAuthorization
+    public Response post(@Context ContainerRequestContext context,
+                         @ApiParam @HeaderParam(APP_KEY) String token,
+                         @ApiParam @NotNull @Valid BroadcastMessage message) {
+        try {
+            trace(message);
+
+            final UUID providerId = (UUID) context.getProperty(PROVIDER_ID);
+
+            List<UUID> botIds = botsDAO.getBotIds(providerId);
+
+            final UUID broadcastId = UUID.randomUUID();
+
+            MDCUtils.put("broadcastId", broadcastId);
+            Logger.info("BroadcastV2Resource.post: `%s`", message.mimeType);
+
+            for (UUID botId : botIds) {
+                broadcast.submit(() -> {
+                    if (message.mimeType.startsWith("audio")) {
```

Comment:
maybe we can add text as well? then we can easily migrate to this new version when it is properly tested